Shared/ReflectionHelper.cs

```diff
@@ -1913,6 +1909,21 @@ public static T PopAt<T>(this IList<T> list, int index)
                 yield return y;
             }
         }
+
+        public static IEnumerable<(T1, T2)> WithContext<T1, T2>(this IEnumerable<T1> src, Func<T2> ctx)
+        {
+            var e = src.GetEnumerator();
+            if (e.MoveNext() == false)
+                    yield break;
+            var ctxValue = ctx();
+            while (true)
+            {
+                yield return (e.Current, ctxValue);
+                if (e.MoveNext() == false)
+                    break;
+            }
+        } 
+        
```

Comment:
It doesn't look like this is used anywhere, and I don't understand how it's useful?
```suggestion
```

Reply:
👍 lets delete that!